extensions/amp-consent/0.1/consent-config.js

```diff
@@ -0,0 +1,219 @@
+/**
+ * Copyright 2018 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {CMP_CONFIG} from './cmps';
+import {CONSENT_POLICY_STATE} from '../../../src/consent-state';
+import {deepMerge, dict} from '../../../src/utils/object';
+import {dev, user} from '../../../src/log';
+import {getChildJsonConfig} from '../../../src/json';
+import {isExperimentOn} from '../../../src/experiments';
+import {toWin} from '../../../src/types';
+
+const TAG = 'amp-consent/consent-config';
+
+export class ConsentConfig {
+
+  /** @param {!Element} element */
+  constructor(element) {
+    /** @private {!Element} */
+    this.element_ = element;
+
+    /** @private {!Window} */
+    this.win_ = toWin(element.ownerDocument.defaultView);
+
+    /** @private {boolean} */
+    this.isMultiSupported_ = isExperimentOn(this.win_, 'multi-consent');
+
+    /** @private {?JsonObject} */
+    this.config_ = null;
+  }
+
+  /**
+   * Read validate and return the config
+   * @return {!JsonObject}
+   */
+  getConfig() {
+    if (!this.config_) {
+      this.config_ = this.validateAndParseConfig_();
+    }
+    return this.config_;
+  }
+
+  /**
+   * Read and parse consent config
+   * An example valid config json looks like
+   * {
+   *   "consents": {
+   *     "consentABC": {
+   *       "checkConsentHref": "https://fake.com"
+   *     }
+   *   }
+   * }
+   * @return {!JsonObject}
+   */
+  validateAndParseConfig_() {
+    const inlineConfig = this.getInlineConfig_();
+
+    const cmpConfig = this.getCMPConfig_();
+
+    const config = /** @type {!JsonObject} */
```

Comment:
Should we be merging both? I feel like this may lead to unexpected behavior. If this is alreayd how v1 works then no worries. But I think it would be better to just log an error if a developer has both.

Reply:
I just noticed that we had a multi-consent experiment. Perhaps you can explain the idea behind this to me?